pkg/apis/numaflow/v1alpha1/edge_types.go

```diff
@@ -39,7 +39,16 @@ type Edge struct {
 }
 
 type ForwardConditions struct {
-	KeyIn []string `json:"keyIn" protobuf:"bytes,1,rep,name=keyIn"`
+	Tags *TagConditions `json:"tags" protobuf:"bytes,1,rep,name=tags"`
+}
+
+type TagConditions struct {
+	// Operator specifies the type of operation that should be used for conditional forwarding
+	// value could be "and", "or", "not"
```

Comment:
nit: you can use kubebuilder to validate enum early.
Example: https://github.com/numaproj/numaflow/blob/main/pkg/apis/numaflow/v1alpha1/edge_types.go#L36